spring-kafka/src/main/java/org/springframework/kafka/core/StreamsBuilderFactoryBean.java

```diff
@@ -198,9 +216,15 @@ public synchronized void start() {
 		if (!this.running) {
 			try {
 				Assert.notNull(this.streamsConfig, "'streamsConfig' must not be null");
-				this.kafkaStreams = new KafkaStreams(getObject().build(), this.streamsConfig, this.clientSupplier);
+				Topology topology = getObject().build();
+				if (logger.isDebugEnabled()) {
+					logger.debug(topology.describe());
+				}
+				this.kafkaStreams = new KafkaStreams(topology, this.streamsConfig, this.clientSupplier);
 				this.kafkaStreams.setStateListener(this.stateListener);
-				this.kafkaStreams.setUncaughtExceptionHandler(this.exceptionHandler);
+				this.kafkaStreams.setGlobalStateRestoreListener(this.stateRestoreListener);
+				this.kafkaStreams.setUncaughtExceptionHandler(this.uncaughtExceptionHandler);
+				this.kafkaStreamsCustomizer.customize(this.kafkaStreams);
```

Comment:
@artembilan here one may be overwriting `stateRestoreListener`, `uncaughtExceptionHandler` or `stateListener` via customizer api. Do you think we should warn users about that?